lib/index.js

```diff
@@ -2326,45 +2326,60 @@ Stream.prototype.split = function () {
 exposeMethod('split');
 
 /**
- * Creates a new Stream with the first `n` values from the source.
+ * Creates a new Stream with the values from the source in the range of `start` to `end`.
  *
- * @id take
+ * @id slice
  * @section Transforms
- * @name Stream.take(n)
- * @param {Number} n - integer representing number of values to read from source
+ * @name Stream.slice(start, end)
+ * @param {Number} start - integer representing index to start reading from source
+ * @param {Number} stop - integer representing index to stop reading from source
  * @api public
  *
- * _([1, 2, 3, 4]).take(2) // => 1, 2
+ * _([1, 2, 3, 4]).slice(1, 3) // => 2, 3
  */
+Stream.prototype.slice = function(start, end) {
+    var index = 0;
+    start = typeof start != 'number' ? 0 : start;
+    end = typeof end != 'number' ? Infinity : end;
 
-Stream.prototype.take = function (n) {
-    if (n === 0) {
+    if (start < 0 || start >= end) {
```

Comment:
`start < 0` should be treated as `start === 0`. Like how `drop` handles `n < 0`. Otherwise, `slice(-1, Infinity)` and `drop(-1)` won't the the same.

Also, add a 

```javascript
if (start === 0 && end === Infinity) {
    return this;
}
```

so that we can drop the `if (n <= 0)` check in `drop`.